### ISSUE TITLE
Argument delegation failing in Rails 7

### ISSUE THREAD

**Issue by headius** [jruby/jruby#7174]
Rails 7 sets up `delegate` methods using a function prototype like this:

```ruby
def root(...)
```

This leads to a full trace error like the following (note that the line numbers have been tweaked since the delegation manipulates backtrace location:

```
$ rake importmap:install --trace
["/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/activesupport-7.0.2.3/lib/active_support/core_ext/module/delegation.rb", 171]
rake aborted!
ArgumentError: `root': wrong number of arguments (given 1, expected 0)
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/activesupport-7.0.2.3/lib/active_support/core_ext/module/delegation.rb:238:in `root'
org/jruby/RubyKernel.java:2010:in `public_send'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/railties-7.0.2.3/lib/rails/railtie.rb:224:in `method_missing'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/actiontext-7.0.2.3/lib/action_text/engine.rb:17:in `<class:Engine>'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/actiontext-7.0.2.3/lib/action_text/engine.rb:11:in `<module:ActionText>'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/actiontext-7.0.2.3/lib/action_text/engine.rb:10:in `<main>'
org/jruby/RubyKernel.java:1017:in `require'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/railties-7.0.2.3/lib/rails/all.rb:20:in `block in <main>'
org/jruby/RubyArray.java:1978:in `each'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/railties-7.0.2.3/lib/rails/all.rb:18:in `<main>'
org/jruby/RubyKernel.java:1017:in `require'
/Users/headius/projects/jruby/testapp/config/application.rb:3:in `<main>'
org/jruby/RubyKernel.java:1017:in `require'
org/jruby/RubyKernel.java:1045:in `require_relative'
/Users/headius/projects/jruby/testapp/Rakefile:4:in `<main>'
org/jruby/RubyKernel.java:1052:in `load'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/rake_module.rb:29:in `load_rakefile'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/application.rb:710:in `raw_load_rakefile'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/application.rb:104:in `block in load_rakefile'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/application.rb:103:in `load_rakefile'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/application.rb:82:in `block in run'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/lib/rake/application.rb:80:in `run'
/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rake-13.0.6/exe/rake:27:in `<main>'
org/jruby/RubyKernel.java:1052:in `load'
/Users/headius/projects/jruby/bin/rake:23:in `<main>'
```

This could be broken in multiple different locations:

* the `method_missing` in this trace may not be forwarding arguments correctly
* some other caller may be handling the delegated `root` method incorrectly
* the delegation in `root` may be broken, incorrectly passing arguments on

**Comment by headius** [comment]
Likely to be fixed by #7180.

**Comment by enebo** [comment]
I think there will still be some errors since I see a couple of arity issues while trying rails 7 unit tests but this is "fixed".  The things which still exist will get made into tests/specs and fixed independently of #7180 which fixed known issues.
